### PR TITLE
[Impeller] Gpu model information to Skia gold

### PR DESCRIPTION
**Pull request by dnfield**
Does two things:

- Exposes a string about the GPU model on `impeller::Context`
- passes that string on to Skia gold when we add a test image.

This should help reduce noise/flakiness in golden images.

### CODE REVIEW

**Review by gaaclarke — APPROVED** [review]
Code lgtm. I don't know how this will manifest in skia gold, but sounds like it should work.

File: testing/skia_gold_client/lib/skia_gold_client.dart [hunk 1]
```diff
@@ -234,6 +236,7 @@ class SkiaGoldClient {
       '--work-dir', _tempPath,
       '--test-name', cleanTestName(testName),
       '--png-file', goldenFile.path,
+      '--add-test-key', 'gpu_string:$gpuString',
```
[the remark this answers is not in the document]
Comment (Piinks) [reply testing/skia_gold_client/lib/skia_gold_client.dart:239]: Is this the same as adding to the list of keys provided as initialization for the test? Those are collected in _getKeys

File: testing/skia_gold_client/lib/skia_gold_client.dart [hunk 2]
```diff
@@ -196,17 +196,18 @@ class SkiaGoldClient {
   Future<void> addImg(
     String testName,
     File goldenFile, {
+    String gpuString = '<unknown>',
```
[the remark this answers is not in the document]
Comment (gaaclarke) [reply testing/skia_gold_client/lib/skia_gold_client.dart]: Done

File: testing/skia_gold_client/lib/skia_gold_client.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaaclarke) [reply testing/skia_gold_client/lib/skia_gold_client.dart]: > Is this the same as adding to the list of keys provided as initialization for the test? Those are collected in _getKeys

Done

**Comment by gaaclarke** [comment]
In local testing the gpu_string is `"Apple M1 Max"`, so that sounds good.

I'm going to rebase to clear up the validation problem.

**Review by mdebbar — APPROVED** [review]
The dimensions change looks good to me!

**Pull request merged**